zkevm-circuits/src/pi_circuit.rs

```diff
@@ -1612,30 +1815,69 @@ impl<F: Field> SubCircuit<F> for PiCircuit<F> {
                         0,
                         0,
                         0u8,
-                        &mut rpi_bytes_keccakrlc,
+                        &mut rpi_bytes_keccak_rlc,
                         challenges,
-                        &mut current_offset,
+                        &mut current_rpi_offset,
                         &mut rpi_bytes,
                         false,
                         F::ZERO,
                         zero_cell.clone(),
                     )?;
                     call_data_offset += 1;
                 }
-                assert_eq!(current_offset, 0);
+                assert_eq!(
+                    start_offset - current_rpi_offset,
+                    N_BYTES_ONE
+                        + N_BYTES_BLOCK
+                        + N_BYTES_EXTRA_VALUE
+                        + Self::Config::circuit_len_tx_id(config.max_txs)
+                        + Self::Config::circuit_len_tx_index(config.max_txs)
+                        + Self::Config::circuit_len_tx_values(config.max_txs)
+                        + config.max_calldata
+                );
+
+                // lookup assignment
+                // also assign empty to last of TxTable
+                config.assign_empty_txtable_row(&mut region, call_data_offset)?;
+
+                // assign withdrawal table and padding rows
+                let mut withdrawal_offset = 0;
+                let wd_default = Withdrawal::default();
+                iter::empty()
+                    .chain(&self.public_data.withdrawals)
+                    .chain(
+                        (0..(config.max_withdrawals - self.public_data.withdrawals.len()))
+                            .map(|_| &wd_default),
+                    )
+                    .enumerate()
+                    .try_for_each(|(_, wd)| -> Result<(), Error> {
+                        config.assign_wd_table_row(
+                            &mut region,
+                            withdrawal_offset,
+                            wd,
+                            &mut rpi_bytes_keccak_rlc,
+                            challenges,
+                            &mut current_rpi_offset,
+                            &mut rpi_bytes,
+                            zero_cell.clone(),
+                        )?;
+                        withdrawal_offset += 1;
+                        Ok(())
+                    })?;
+                assert_eq!(current_rpi_offset, 0);
+
+                config.assign_empty_wd_table_row(&mut region, withdrawal_offset)?;
```

Comment:
Just a double confirm ;) 
Do we need to assign empty row in wd_table? I guess we dont need because there is no constraints applied on wd_table to access rotation(1)

Reply:
you're right! Forgot to remove it. Fixed in [bbaa6ae](https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/1634/commits/bbaa6ae4c184d125c9cb428a64b21a92f87e0138)